### PR TITLE
BE-1232 Added run-cloud command

### PR DESCRIPTION
**Pull request by xzandrew**
Added run-cloud command for running in AWS. Stores cache in S3 and uses JSON config passed through the command line.

### CODE REVIEW

**Comment by xzandrew** [comment]
Should be ok to merge.

File: commands/cconfig/cloud.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+	var obj struct {
+		Integrations map[string]struct {
+			//Location string `json:"location"`
+			Authorization struct {
+				AccessToken  string `json:"access_token"`
+				RefreshToken string `json:"refresh_token"`
+			} `json:"authorization"`
+			//Exclusions []string `json:"exclusions"`
+		} `json:"integrations"`
```
Comment (jhaynie) [inline commands/cconfig/cloud.go]: we haven't added exclusions support but that will come in here in the next day or so.

File: commands/cconfig/config.go [hunk 2]
```diff
@@ -1,34 +1,73 @@
...
+	File() File
+}
+
+type Service struct {
+	Name                 string   `yaml:"name"`
+	URL                  string   `yaml:"url"`
+	APIKey               string   `yaml:"apikey"`
+	Username             string   `yaml:"username"`
+	Repos                []string `yaml:"repos"`
+	MaxExportConcurrency int      `yaml:"max-export-concurrency"`
```
Comment (jhaynie) [inline commands/cconfig/config.go]: this won't work for cloud. you'll need access_token and refresh_token

File: commands/cconfig/cloud.go [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+		a := s.services[i]
+		b := s.services[j]
+		return a.Name < b.Name
+	})
+
+	var viperServices []interface{}
+	for name, in := range obj.Integrations {
+		res := map[interface{}]interface{}{}
+		res["name"] = name
+		res["apitoken"] = in.Authorization.AccessToken
```
Comment (xzandrew) [inline commands/cconfig/cloud.go]: I'm setting apitoken and refreshtoken here.

File: commands/cconfig/config.go (hunk 2 above)
Comment (xzandrew) [reply commands/cconfig/config.go]: These values are only used for OnPremise config and the whole struct was copied from existing code. The access token and refresh token are set in viper services config.

File: commands/cconfig/cloud.go (hunk 1 above)
Comment (xzandrew) [reply commands/cconfig/cloud.go]: I'll add the code to use the exclusions then. Are these the ids of repos to skip?

**Pull request merged**